Core/src/org/sleuthkit/autopsy/mainui/datamodel/AnalysisResultDAO.java

```diff
@@ -642,75 +686,191 @@ private int compareSetStrings(String a, String b) {
+                        BlackboardAttribute setAttr = art.getAttribute(BlackboardAttribute.Type.TSK_SET_NAME);
+                        String setName = setAttr == null ? null : setAttr.getValueString();
+                        setMap.computeIfAbsent(Pair.of(art.getType(), setName), (k) -> new HashSet<>())
+                                .add(art.getDataSourceObjectID());
+                        
+                    } else if (BlackboardArtifact.Category.ANALYSIS_RESULT.equals(art.getType().getCategory())) {
+                        analysisResultMap.computeIfAbsent(art.getType(), (k) -> new HashSet<>())
+                                .add(art.getDataSourceObjectID());
+                    }
+                } catch (TskCoreException ex) {
+                    logger.log(Level.WARNING, "Unable to fetch necessary information for artifact id: " + art.getId(), ex);
+                }
+            }
+        }
+
+        // don't continue if no relevant items found
+        if (analysisResultMap.isEmpty() && setMap.isEmpty() && keywordHitsMap.isEmpty()) {
+            return Collections.emptyList();
+        }
+        
+        clearRelevantCacheEntries(analysisResultMap, setMap);
+        
+        List<AnalysisResultEvent> daoEvents = getDAOEvents(analysisResultMap, setMap);
+        Collection<TreeEvent> treeEvents = this.treeCache.enqueueAll(daoEvents).stream()
+                .map(arEvt -> getTreeEvent(arEvt, false))
+                .collect(Collectors.toList());
+        
+        return Stream.of(daoEvents, treeEvents)
+                .flatMap(lst -> lst.stream())
+                .collect(Collectors.toList());
+    }
 
     /**
-     * Handles basic functionality of fetching and paging of analysis results.
+     * Generate DAO events from digest of Autopsy events.
+     *
+     * @param analysisResultMap Contains the analysis results that do not use a
+     *                          set name. A mapping of analysis result type ids
+     *                          to data sources where the results were created.
+     * @param resultsWithSetMap Contains the anlaysis results that do use a set
+     *                          name. A mapping of (analysis result type id, set
+     *                          name) to data sources where results were
+     *                          created.
+     *
+     * @return The list of dao events.
      */
-    static abstract class AbstractAnalysisResultFetcher<T extends AnalysisResultSearchParam> extends DAOFetcher<T> {
-
-        /**
-         * Main constructor.
-         *
-         * @param params Parameters to handle fetching of data.
-         */
-        public AbstractAnalysisResultFetcher(T params) {
-            super(params);
-        }
+    private List<AnalysisResultEvent> getDAOEvents(Map<BlackboardArtifact.Type, Set<Long>> analysisResultMap, Map<Pair<BlackboardArtifact.Type, String>, Set<Long>> resultsWithSetMap) {
```

Comment:
This seems to be only getting the "Table" events and we later determine the "Tree" events. WE should rename this method/comments to make it clear that it's only Table and not all events.